lib/Result.ts

```diff
@@ -60,6 +60,28 @@ export class SuccessResult<Success, Failure> {
     return this
   }
 
+  /**
+   * Runs the handler with the current success value and passes the current success value through
+   * if the handler is successful.
+   *
+   * @param handler a function to process the current success value and either passthrough the value or return a failure.
+   */
+  passthroughSuccess<NewSuccess, NewFailure>(
+    handler: (value: Success) => AwaitableResult<NewSuccess, NewFailure>
+  ): SuccessResult<Success, NewFailure | Failure> {
```

Comment:
```suggestion
  ): AwaitableResult<Success, NewFailure | Failure> {
```
The handler may return a failure or promise result.

Reply:
fixed